src/adapters/rhythmone.js

```diff
@@ -0,0 +1,496 @@
+        
+        img.src = url;
+      }
+    
+      function frameWrap(target, markup, url){
+        var f = document.createElement("iframe");
+        f.style.border="0";
+        f.scrolling="no";
+        f.seamless="seamless";
+        f.style.height = bidResponse.h+"px";
+        f.style.width = bidResponse.w+"px";
+        
+        target.style.height = bidResponse.h+"px";
+        target.style.width = bidResponse.w+"px";
+        target.appendChild(f);
+        
+        if(markup){
+          f.contentWindow.document.open();
+          f.contentWindow.document.write("<html><head></head><body style=\"margin:0;padding:0;\">"+markup+"</body></html>");
+          if((/(MSIE|Trident|Edge)/).test(window.navigator.userAgent) === false)
+            f.contentWindow.document.close();
+        }
+        else if(url)
+          f.src = url;
+      }
+    
+      var w = window,
+        pjs = (function(){
+          try{
+            while(w){
+              if(w.$$PREBID_GLOBAL$$ || w === window.top) return w.$$PREBID_GLOBAL$$;
+              w = w.parent;
+            }
+          }
+          catch(ex){}
+          return null;
+        })(),
+        responses = (pjs ? pjs.getBidResponses() : {}),
+        slotBids = responses[target],
+        secondPrice = 0;
+    
+      //console.log(responses);
+    
+      log(responses, publisherId, pjs, w);
+      
+      var prices = [bidRequest.imp[0].bidfloor];
+        
+      if(slotBids)
+        for(var i=0; i<slotBids.bids.length; i++)
+          prices.push(parseFloat(slotBids.bids[i].cpm));
+      
+      prices.sort();
+      secondPrice = prices[prices.length-2]+0.01;
+    
+      var macroPattern = /\$\{AUCTION_([A-Z_]+)\}/g,
+        values = {
+          ID: bidRequest.id,
+          BID_ID: bidResponse.id,
+          IMP_ID: bidResponse.impid,
+          SEAT_ID: seat.id,
+          AD_ID: bidResponse.adid,
+          PRICE: secondPrice,
+          CURRENCY: bidRequest.cur[0]
+        };
+      
+      if(markup)
+        markup = markup.replace(macroPattern, function(match){
+          var x = macroPattern.exec(match)[1];
+          //console.log("found macro "+match+" replacing with value "+values[x]);
+          if(values[x]) return values[x];
+          return match;
+        });
+      
+      if(url)
+        url = url.replace(macroPattern, function(match){
+          var x = macroPattern.exec(match)[1];
+          //console.log("found macro "+match+" replacing with value "+values[x]);
+          if(values[x]) return values[x];
+          return match;
+        });
+      
+      frameWrap(document.getElementById(target+"_wrapper"), markup, url);
+    }
+    */
+  };
+  
+  //console.log(JSON.stringify(rhythmBidderUtilities.process.toString()));
+  
+  function load( url, jsonObject, callback){
+    
+    var json = JSON.stringify(jsonObject),
+      x;
+    
+    logToConsole("posting "+json+" to "+url);
+    
+    if(typeof global.XDomainRequest !== "undefined"){
```

Comment:
We have utility for making ajax requests - please use `https://github.com/prebid/Prebid.js/blob/master/src/ajax.js` instead